extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java

```diff
@@ -353,42 +354,57 @@ private synchronized HueBridgeHandler getHueBridgeHandler() {
 
     @Override
     public void onLightStateChanged(HueBridge bridge, FullLight fullLight) {
-        if (fullLight != null && fullLight.getId().equals(lightId)) {
+        logger.trace("onLightStateChanged() was called");
 
-            initializeProperties();
+        if (fullLight == null) {
```

Comment:
Might it make sense to use the new `@NonNull` annotation on the parameter here, so that this check becomes obsolete?